### DISCUSSION TITLE
Unable to access nbhits

### DISCUSSION THREAD

**Discussion by JSShant** [algolia/instantsearch#5833] [Q&A] [answered]
Im searching an index with particular filters applied to it as such:
![image](https://github.com/algolia/instantsearch/assets/105369293/514f5a21-206e-4ff2-a8cf-79ae8d3705d7)
Im getting the nbHits from lotHits:
![image](https://github.com/algolia/instantsearch/assets/105369293/20a3e659-d84c-4738-8aa9-39f239e0679d)
However im not able to access it like : lotResultsQuery.data?.results[0].nbHits 
This throws an error: Property 'nbHits' does not exist on type 'SearchForFacetValuesResponse | SearchResponse<unknown>'.
  Property 'nbHits' does not exist on type 'SearchForFacetValuesResponse'.
In fact when trying to access the values im only being prompted for two options:
![image](https://github.com/algolia/instantsearch/assets/105369293/1baa911f-3359-46bc-b1e8-98c7058915b8)

**Comment by Haroenv** [comment, marked as answer]
This is actually related to the search client, not InstantSearch, but it's because `results` could be of type `default` or `facet`, you can cast it: `guardSearchResponse(lotResultsQuery.data?.results[0])` with

```ts
// algoliasearch v4
import { SearchResponse, MultipleQueriesResponse } from "algoliasearch";

function guardSearchResponse<TObject>(
  result: MultipleQueriesResponse<TObject>['results'][number]
): result is SearchResponse<TObject> {
  return 'hits' in result;
}
```


```ts
// algoliasearch v5
import type { SearchResponse, SearchResult } from 'algoliasearch';

function guardSearchResponse<TObject>(
  result: SearchResult
): result is SearchResponse<TObject> {
  return 'hits' in result;
}
```

**Reply by nikoewe** [reply]
this does not work, the export MultipleQueriesResponse doesn't exist

**Reply by Haroenv** [reply]
you're responding on a 2023 thread, the type has changed since @nikoewe. This is the v5 version:

```ts
import type { SearchResponse, SearchResult } from 'algoliasearch';

function guardSearchResponse<TObject>(
  result: SearchResult
): result is SearchResponse<TObject> {
  return 'hits' in result;
}
```

**Reply by nikoewe** [reply]
Damn, haven't seen that, sorry. Thanks!
